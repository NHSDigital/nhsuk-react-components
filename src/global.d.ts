declare module 'nhsuk-frontend/packages/components/header/header.js';
declare module 'nhsuk-frontend/packages/components/checkboxes/checkboxes.js';
declare module 'nhsuk-frontend/packages/components/character-count/character-count.js';
declare module 'nhsuk-frontend/packages/components/tabs/tabs.js';
