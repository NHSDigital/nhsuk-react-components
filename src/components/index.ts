export * from './content-presentation/index.js';
export * from './form-elements/index.js';
export * from './layout/index.js';
export * from './navigation/index.js';
export * from './typography/index.js';
export * from './utils/index.js';
