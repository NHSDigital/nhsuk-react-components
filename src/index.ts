export * from './components';
export * from './patterns';
