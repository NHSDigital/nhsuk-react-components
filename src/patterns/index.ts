export * from './nav-a-z/index.js';
export * from './review-date/index.js';
