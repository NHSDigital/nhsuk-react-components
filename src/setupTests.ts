import { TextDecoder, TextEncoder } from 'util';
import '@testing-library/jest-dom';
import { outdent } from 'outdent';

/**
 * Polyfill TextEncoder/TextDecoder for ReactDOM
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(globalThis, {
  TextEncoder: {
    value: TextEncoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
  TextDecoder: {
    value: TextDecoder,
    configurable: true,
    enumerable: false,
    writable: true,
  },
});

/**
 * Polyfill Element methods for NHS.UK frontend error summary
 *
 * @see {@link https://github.com/jsdom/jsdom/issues/1695}
 */
Object.defineProperties(Element.prototype, {
  scrollIntoView: { value: jest.fn() },
});

/**
 * Polyfill `matchMedia()` for NHS.UK frontend tabs
 */
Object.defineProperty(globalThis, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: true,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    addEventListener: jest.fn(),
    removeListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

beforeEach(() => {
  const stylesheet = document.createElement('style');

  stylesheet.innerHTML = outdent`
    :root {
      --nhsuk-breakpoint-mobile: 20rem;
      --nhsuk-breakpoint-tablet: 40.0625rem;
      --nhsuk-breakpoint-desktop: 48.0625rem;
      --nhsuk-breakpoint-large-desktop: 61.875rem;
    }
  `;

  // Add styles for NHS.UK frontend checks
  document.head.appendChild(stylesheet);

  // Flag NHS.UK frontend as supported
  document.body.classList.add('nhsuk-frontend-supported');
});
