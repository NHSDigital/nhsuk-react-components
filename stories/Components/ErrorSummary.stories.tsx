import React from 'react';
import { ErrorSummary } from '../../src';
import { Meta, StoryObj } from '@storybook/react';

/**
 * This component can be found in the `nhsuk-frontend` repository <a href="https://github.com/nhsuk/nhsuk-frontend/tree/master/packages/components/error-summary" target="_blank" rel="noopener noreferrer">here</a>.
 *
 * ## Implementation Notes
 *
 * The `ErrorSummary` component has four subcomponents:
 *
 * - `ErrorSummary.Title`
 * - `ErrorSummary.Body`
 * - `ErrorSummary.List`
 * - `ErrorSummary.Item`
 *
 * ## Usage
 *
 * ### Standard
 *
 * ```jsx
 * import { ErrorSummary } from "nhsuk-react-components";
 *
 * const Element = () => {
 *     return (
 *         <ErrorSummary aria-labelledby="error-summary-title" role="alert" tabIndex={-1}>
 *             <ErrorSummary.Title id="error-summary-title">There is a problem</ErrorSummary.Title>
 *             <ErrorSummary.Body>
 *                 <p>Optional description of the errors and how to correct them</p>
 *                 <ErrorSummary.List>
 *                     <ErrorSummary.Item href="#example-error-1">
 *                         Link to error with explanation
 *                     </ErrorSummary.Item>
 *                     <ErrorSummary.Item href="#example-error-2">
 *                         Link to error with explanation
 *                     </ErrorSummary.Item>
 *                 </ErrorSummary.List>
 *             </ErrorSummary.Body>
 *         </ErrorSummary>
 *     );
 * }
 * ```
 */
const meta: Meta<typeof ErrorSummary> = {
  title: 'Components/ErrorSummary',
  component: ErrorSummary,
};
export default meta;
type Story = StoryObj<typeof ErrorSummary>;

ErrorSummary.Title.displayName = 'ErrorSummary.Title';
ErrorSummary.Body.displayName = 'ErrorSummary.Body';
ErrorSummary.List.displayName = 'ErrorSummary.List';
ErrorSummary.Item.displayName = 'ErrorSummary.Item';

export const Standard: Story = {
  render: (args) => (
    <ErrorSummary aria-labelledby="error-summary-title" role="alert" tabIndex={-1}>
      <ErrorSummary.Title id="error-summary-title">There is a problem</ErrorSummary.Title>
      <ErrorSummary.Body>
        <p>Optional description of the errors and how to correct them</p>
        <ErrorSummary.List>
          <ErrorSummary.Item href="#example-error-1">
            Link to error with explanation
          </ErrorSummary.Item>
          <ErrorSummary.Item href="#example-error-2">
            Link to error with explanation
          </ErrorSummary.Item>
        </ErrorSummary.List>
      </ErrorSummary.Body>
    </ErrorSummary>
  ),
};
