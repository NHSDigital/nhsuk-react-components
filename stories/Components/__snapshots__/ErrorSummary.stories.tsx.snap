// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ErrorSummary Standard smoke-test 1`] = `
<div class="nhsuk-error-summary"
     aria-labelledby="error-summary-title"
     role="alert"
     tabindex="-1"
>
  <h2 class="nhsuk-error-summary__title"
      id="error-summary-title"
  >
    There is a problem
  </h2>
  <div class="nhsuk-error-summary__body">
    <p>
      Optional description of the errors and how to correct them
    </p>
    <ul class="nhsuk-list nhsuk-error-summary__list">
      <li>
        <a href="#example-error-1">
          Link to error with explanation
        </a>
      </li>
      <li>
        <a href="#example-error-2">
          Link to error with explanation
        </a>
      </li>
    </ul>
  </div>
</div>
`;
